bin/generate.py

```diff
@@ -1206,6 +1229,23 @@ def __init__(self, problem, key, name: str, yaml: dict = None, parent=None):
                             self.path,
                         )
 
+    @overload
```

Comment:
Add a comment what we're looking at here.

Reply:
The `@overload` definitions are basically two "versions" of the static types, and writing them like this was easier than trying to (ab)use generic types. I've added the comments, let me know if this clears things up :slightly_smiling_face: